Source/WTF/wtf/SequesteredImmortalHeap.h

```diff
@@ -94,6 +94,118 @@ class ConcurrentDecommitQueue {
     Lock m_decommitLock { };
 };
 
+// FIXME: a lot of this, but not all, can be de-duped with SequesteredArenaAllocator::Arena
+class SequesteredImmortalAllocator {
+    constexpr static bool verbose { false };
+    constexpr static size_t minGranuleSize { 16 * KB };
+    constexpr static size_t minHeadAlignment { alignof(std::max_align_t) };
+public:
+    SequesteredImmortalAllocator() = default;
+
+    SequesteredImmortalAllocator(SequesteredImmortalAllocator&& other) = delete;
+    SequesteredImmortalAllocator& operator=(SequesteredImmortalAllocator&& other) = delete;
+    SequesteredImmortalAllocator(const SequesteredImmortalAllocator&) = delete;
+    SequesteredImmortalAllocator& operator=(const SequesteredImmortalAllocator&) = delete;
+
+    void* allocate(size_t bytes)
+    {
+        void* retval { };
+        void* newAllocHead { };
+        {
+            Locker lock(m_lock);
+            retval = allocateImpl(bytes);
+            newAllocHead = reinterpret_cast<void*>(m_allocHead);
+        }
+        dataLogLnIf(verbose,
+            "SequesteredImmortalAllocator at ", RawPointer(this),
+            ": allocated ", bytes, "B: alloc (", RawPointer(retval),
+            "), allocHead (", RawPointer(newAllocHead),
+            ")");
+        return retval;
+    }
+
+    void* alignedAllocate(size_t alignment, size_t bytes)
+    {
+        void* retval { };
+        void* newAllocHead { };
+        {
+            Locker lock(m_lock);
+            retval = alignedAllocateImpl(alignment, bytes);
+            newAllocHead = reinterpret_cast<void*>(m_allocHead);
+        }
+        dataLogLnIf(verbose,
+            "SequesteredImmortalAllocator at ", RawPointer(this),
+            ": align-allocated ", bytes, "B: alloc (", RawPointer(retval),
+            "), allocHead (", RawPointer(newAllocHead),
+            ")");
+        return retval;
+    }
+private:
+    uintptr_t headIncrementedBy(size_t bytes) const
+    {
+        const size_t alignmentMask = minHeadAlignment - 1;
+        return (m_allocHead + bytes + alignmentMask) & ~alignmentMask;
+    }
+
+    uintptr_t headAlignedUpTo(size_t alignment) const
```

Comment:
I think this whole function can probably be replaced by `WTF::roundUpToMultipleOf`.